pkg/cli/cli.go

```diff
@@ -144,8 +145,21 @@ func newCLI(options ...Option) (*CLI, error) {
 func (c *CLI) buildCmd() error {
 	c.cmd = c.newRootCmd()
 
+	var uve config.UnsupportedVersionError
+
 	// Get project version and plugin keys.
-	if err := c.getInfo(); err != nil {
+	switch err := c.getInfo(); {
+	case err == nil:
+	case errors.As(err, &uve) && uve.Version.Compare(config.Version{Number: 3, Stage: stage.Alpha}) == 0:
+		// Check if the corresponding stable version exists, set c.projectVersion and break
+		stableVersion := config.Version{
+			Number: uve.Version.Number,
+		}
+		if config.IsRegistered(stableVersion) {
+			// Use the stableVersion
+			c.projectVersion = stableVersion
+		}
```

Comment:
I think this is my last comment. Shouldn't it return an error if the stable wasn't found?
```suggestion
		} else {
			return err
		}
```

Reply:
@Adirio I don't mind the comments, I really found them valuable.

Reply:
Returned error and added a test case for that condition.